lib/puppet/type/rabbitmq_user.rb

```diff
@@ -40,20 +40,12 @@
   newproperty(:password) do
     desc 'User password to be set *on creation* and validated each run'
     def insync?(_is)
-      provider.check_password
-    end
-
-    def set(_value)
-      provider.change_password
+      provider.check_password(should)
     end
 
     def change_to_s(_current, _desired)
       'password has been changed'
     end
-
-    def should_to_s(_newvalue = @should)
-      '<new password>'
-    end
   end
 
   newproperty(:admin) do
```

Comment:
You can use ```newproperty(:admin, boolean: true, parent: Parent::Property::Boolean)``` and remove the body (except ```desc```). Then you can use true booleans in the provider. You might need a ```require 'puppet/property/boolean``` but I'm not sure about that.

Reply:
I think this is a good idea, but I feel like it should be implemented separately, and then we can review all the type parameters at once, and adjust any that we can switch to true booleans / integers / etc in one 'breaking' changeset.

Reply:
That's fair

Reply:
It seems like the existing code is (in some cases) as simple as:
```
  newparam(:force) do
    defaultto(:false)
    newvalues(:true, :false)
  end
```
which actually seems simpler than the suggested code in https://docs.puppet.com/puppet/5.1/custom_types.html, and without that extra `require`. Also, the puppet boolean type allows `'true'` `yes`, and various other things, which seems kind of regressive.